ZuluContent/Items/Misc/OstardEgg.cs

```diff
@@ -0,0 +1,116 @@
+﻿using Server.Mobiles;
+using Server.Spells;
+using System;
+
+namespace Server.Items
+{
+
+    public class OstardEgg : Item
+    {
+        public override double DefaultWeight => 0.02;
+        public static WeightedRandomType<BaseCreature> WeightedOstards;
+
+        [Constructible]
+        public OstardEgg() : this(1)
+        {
+        }
+
+
+        [Constructible]
+        public OstardEgg(int amountFrom, int amountTo) : this(Utility.RandomMinMax(amountFrom, amountTo))
+        {
+        }
+
+
+        [Constructible]
+        public OstardEgg(int amount) : base(0x1725)
+        {
+            Stackable = true;
+            Amount = amount;
+            Name = "Ostard Egg";
+            Hue = 1102;
+        }
+
+        [Constructible]
+        public OstardEgg(Serial serial) : base(serial)
+        {
+        }
+
+        public override void OnDoubleClick(Mobile from)
+        {
+            if (!Movable)
+                return;
+
+            if (!IsChildOf(from.Backpack))
+            {
+                from.SendLocalizedMessage(1042001); // That must be in your pack for you to use it.
+                return;
+            }
+            
+            var p = new Point3D(from);
+            
+            if (!SpellHelper.FindValidSpawnLocation(from.Map, ref p, true))
+                return;
+
+            Consume(1);
+            from.SendMessage("The egg begins to move and ");
+
+            var ostard = GetOstardByChance();
+
+            if (from.Skills.AnimalTaming.Value > 90 && Utility.RandomMinMax(0, 100) > 70)
+            {
+                ostard.Owners.Add(from);
+                ostard.SetControlMaster(from);
+                from.SendMessage("A baby ostard appears and accepts you as his master!");
+            }
+            else
+                from.SendMessage("A baby ostard appears!");
+
+            ostard.MoveToWorld(p, from.Map);            
+        }
+
+        public override void Serialize(IGenericWriter writer)
+        {
+            base.Serialize(writer);
+
+            writer.Write((int)0); // version
+        }
+
+        public override void Deserialize(IGenericReader reader)
+        {
+            base.Deserialize(reader);
+
+            int version = reader.ReadInt();
+        }
+
+        static BaseCreature GetOstardByChance()
```

Comment:
Think you might have gotten a little confused here, this is still doing the same thing but in a static method! You need to use a **Static Constructor**, which is a special static method you use to initialize static fields on your class.

>A static constructor is used to initialize any static data, or to perform a particular action that needs to be performed once only. It is called automatically before the first instance is created or any static members are referenced.
https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/static-constructors

You would refactor it to something like this:

![image](https://user-images.githubusercontent.com/1094679/108790336-8cc06b00-75d0-11eb-9234-e70d22eca0bc.png)